src/libmeasurement_kit/ooni/orchestrate.cpp

```diff
@@ -157,55 +158,62 @@ void Client::register_probe(std::string &&password,
                             Callback<Error &&, Auth &&> &&cb) const {
     // Copy the data contained by this object so we completely detach the
     // destiny of `this` and of the callback.
-    AsyncRunner::global()->start_("orchestrate::register_probe", logger, [
+    //
+    // We must copy or move everything into the initial lambda and then from
+    // there the task is synchronous because run_...() is blocking.
+    Worker::default_tasks_queue()->run_in_background_thread([
         password = std::move(password), meta = *this, cb = std::move(cb)
-    ](Continuation<> && done) mutable {
-        do_register_probe(
-              std::move(password), meta, AsyncRunner::global()->reactor(),
-              [ done = std::move(done),
-                cb = std::move(cb) ](Error && error, Auth && auth) mutable {
-                  done([
-                      error = std::move(error), cb = std::move(cb),
-                      auth = std::move(auth)
-                  ]() mutable { cb(std::move(error), std::move(auth)); });
-              });
+    ]() mutable {
+        Var<Reactor> reactor = Reactor::make();
+        reactor->run_with_initial_event([&]() {
```

Comment:
This is basically the same as we do for running tests. Here in orchestration it has not been factored in a common function because in here it is actually a couple of lines of code, so I didn't see the need.